File: Design Patterns/Strategy Pattern/Java/com.faculdadevincit.designpatterns.strategypattern/src/PagamentoPix.java
public class PagamentoPix implements IPagamentoStrategy{

    @Override
    public void pagar(double valor) {
        double desconto = valor *0.10;
        double valorComDesconto = valor - desconto;
        System.out.println("Pagamento de R$" + valorComDesconto + " em PIX realizado com sucesso!");
        System.out.println("Desconto no valor de R$" + desconto);
    }

}

File: Design Patterns/Strategy Pattern/CSharp/FaculdadeVincit.DesignPattern.StrategyPattern/PagamentoPix.cs
namespace FaculdadeVincit.DesignPattern.StrategyPattern;

public class PagamentoPix : IPagamentoStrategy
{
    public void Pagar(double valor)
    {
        var desconto = valor *0.10;
        var valorComDesconto = valor - desconto;
        System.Console.WriteLine($"Pagamento de R$ {valorComDesconto} por PIX recebido com sucesso!");
        System.Console.WriteLine($"Desconto por pagamento à vista: R$ {desconto}");
    }

}
